### PR TITLE
Roll command

### PR DESCRIPTION
**Pull request by bramz**

### CODE REVIEW

**Comment by RaasAhsan** [comment]
Roll expressions:
```
3
3d5
6d9+4
2d4+3d6+4+9
3 + 3d5 + 8
```

Roll command:
```
!roll <roll expression>
```

**Comment by RaasAhsan** [comment]
screen shot 2019-02-27 at 2 26 43 am

File: apophis/cmds/roll.py [hunk 1]
```diff
@@ -0,0 +1,208 @@
...
+def add(lhs: Expression, rhs: Expression) -> Expression:
+    return Expression(Add(lhs, rhs))
+
+
+def evaluate(expr: Expression) -> List[int]:
+    value = expr.value
+    if isinstance(value, Constant):
+        return [value.c]
+    elif isinstance(value, Dice):
+        xs = [0] * value.rolls
```
Comment (JosephMoniz) [inline apophis/cmds/roll.py]: You can use `range` instead: https://docs.python.org/3/library/functions.html#func-range

File: apophis/cmds/roll.py [hunk 2]
```diff
@@ -0,0 +1,208 @@
...
+        faces = value.faces
+        terms = map(lambda x: random.randint(1, faces), xs)
+        return list(terms)
```
Comment (JosephMoniz) [inline apophis/cmds/roll.py]: You can merge this line and the previous by using a simple for expression. Something like.
```python
terms = [random.randint(1, faces) for _ in xs]
```

File: apophis/cmds/roll.py [hunk 3]
```diff
@@ -0,0 +1,208 @@
...
+class RollCommand(Command):
+    async def handle(self, context, message):
+        input = message.content[6:].strip()
+        expression = parse(input)
+
+        if expression is None:
+            return await message.channel.send('Invalid roll expression.')
+        else:
+            terms = evaluate(expression)
+            sum_string = ' + '.join('**{}**'.format(str(t)) for t in terms)
```
Comment (JosephMoniz) [inline apophis/cmds/roll.py]: If this is python3 you can use `f-strings` instead of calling `format()`

File: apophis/cmds/roll.py [hunk 4]
```diff
@@ -0,0 +1,208 @@
...
+
+            return await message.channel.send(
+                '{} = **{}**'
+                .format(sum_string, sum(terms))
```
Comment (JosephMoniz) [inline apophis/cmds/roll.py]: If this is python3 you can use `f-strings` instead of calling `format()`

**Pull request merged**